editor/src/components/canvas/dom-walker.ts

```diff
@@ -1178,14 +1189,14 @@ function walkElements(
       additionalElementsToUpdate,
     )
 
-    rootMetadataAccumulator = [...rootMetadataAccumulator, ...collectedMetadata]
+    mergeMetadataMapsWithFragments_MUTATE(rootMetadataAccumulator, collectedMetadata)
     cachedPathsAccumulator = [...cachedPathsAccumulator, ...cachedPaths]
     return {
       rootMetadata: rootMetadataAccumulator,
-      childPaths: collectedMetadata.map((metadata) => metadata.elementPath), // TODO why not extract childPaths from the metadata?
+      childPaths: Object.values(collectedMetadata).map((metadata) => metadata.elementPath), // TODO why not extract childPaths from the metadata?
```

Comment:
is this expensive here? shouldn't `collectMetadata` return a childPaths too?

Reply:
Good idea, see fdf0c24a020642c922d6247af209aa31dad3432a